src/Sarif/CodeGenHints.json

```diff
@@ -1,12 +1,64 @@
 {
+  "annotatedCodeLocation": [
+    {
+      "kind": "BaseTypeHint",
+      "arguments": {
+        "baseTypeNames": [
+          "PropertyBagHolder"
+        ]
+      }
+    }
+  ],
   "AnnotatedCodeLocation.Properties": [
     {
-      "kind": "DictionaryHint"
+      "kind": "DictionaryHint",
+      "arguments": {
+        "valueTypeName": "SerializedPropertyInfo",
+        "namespaceName": "Microsoft.CodeAnalysis.Sarif.Readers",
```

Comment:
`DictionaryHint` now has a `ValueTypeName` property (that thing you were trying to use weeks ago but it didn't exist at the time). That allows the property bags to be of type `IDictionary<string, SerializedPropertyInfo>`.